src/MailboxesParser/ParsedMailbox.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Lukasz
+ * Date: 2017-11-10
+ * Time: 20:09
+ */
+
+namespace Ddeboer\Imap\MailboxesParser;
+
+
+use Ddeboer\Imap\MailboxInterface;
+
+class ParsedMailbox
+{
+    /** @var MailboxInterface */
+    protected $mailbox;
+    protected $order;
+    protected $mailboxName;
+    protected $name;
+    protected $special;
+    protected $delimiter = '.';
+    protected $level = 1;
+    protected $subfolders = 0;
+
+    /**
+     * @return MailboxInterface
+     */
+    public function getMailbox(): MailboxInterface
+    {
+        return $this->mailbox;
+    }
+
+    /**
+     * @param MailboxInterface $mailbox
+     */
+    public function setMailbox(MailboxInterface $mailbox)
```

Comment:
Can we move all the setters to be the constructor?

Reply:
this is internal helper class I see no point.

Reply:
The point is that immutable object are far more easy to maintain.